### DISCUSSION TITLE
Simple way of inlineTool dynamic configuration

### DISCUSSION THREAD

**Discussion by amitrahav** [codex-team/editor.js#1901] [Q&A]
Is there a way of updating inlineTool configuration dynamically?

**Comment by gohabereg** [comment]
Hi there,
Could you provide your use case please?

**Comment by amitrahav** [comment]
I want to write an inline tool that receives list of columns from a google sheet table (that may change), and to be able to select text and tag it with a column name.
So I used a custom configuration for my inline tool, but it can not be reconfigured when my table changes.

**Comment by OneLoneFox** [comment]
I also have a use case for updating a block's configuration dynamically
I propose the addition of an updateConfig method or something for the ToolsAPI
I am writing an editor that has several selectable pages (tabs) in an SPA fashion and I need to update the additionalRequestData config of the editor-js/image plugin which right now I'm doing in a dirty way by literally calling 
`this.editor.configuration.tools.image.config.additionalRequestData.tab_id = this.selectedTab.id;`
It would be nice to have a method for updating a tool's configuration after the initial

**Comment by gohabereg** [comment]
@amitrahav I guess it's not possible at the moment, Inline tools don't have internal state yet

**Comment by gohabereg** [comment]
@OneLoneFox Not sure I got it.
If you have different editor's in different tabs I think it's easier to use several instances

**Comment by OneLoneFox** [comment]
@gohabereg 
> If you have different editor's in different tabs I think it's easier to use several instances

That was the original idea the thing is the tabs can be created and deleted dynamically via API calls and managing several instances of the editor seems a bit heavy for updating a single config value so what I did was create a single instance and every time the user selects a tab fetch the data, clear and update the editor instance's data. But maybe you're right and my approach is not the best.

**Comment by gohabereg** [comment]
I believe several instances is easier to manage and would be logically correct. You can destroy instance if tab is not active (there'll be a lag because of instance loading, so you can do it only if amount of tabs is greater than N)
